front/lib/api/assistant/actions/process.ts

```diff
@@ -0,0 +1,416 @@
+  // later on, the action won't have outputs but the error will be stored on the parent agent
+  // message.
+  const action = await AgentProcessAction.create({
+    relativeTimeFrameDuration: relativeTimeFrame?.duration ?? null,
+    relativeTimeFrameUnit: relativeTimeFrame?.unit ?? null,
+    processConfigurationId: actionConfiguration.sId,
+    schema: actionConfiguration.schema,
+  });
+
+  yield {
+    type: "process_params",
+    created: Date.now(),
+    configurationId: configuration.sId,
+    messageId: agentMessage.sId,
+    dataSources: actionConfiguration.dataSources,
+    action: {
+      id: action.id,
+      type: "process_action",
+      params: {
+        relativeTimeFrame,
+      },
+      schema: action.schema,
+      outputs: null,
+    },
+  };
+
+  const now = Date.now();
+
+  const prompt = await constructPrompt(
+    auth,
+    userMessage,
+    configuration,
+    "Process the retrieved data to extract structured information based on the provided schema."
+  );
+
+  const config = cloneBaseConfig(
+    DustProdActionRegistry["assistant-v2-process"].config
+  );
+
+  // Handle data sources list and parents/tags filtering.
+  config.DATASOURCE.data_sources = actionConfiguration.dataSources.map((d) => ({
+    workspace_id: d.workspaceId,
+    data_source_id: d.dataSourceId,
+  }));
+
+  for (const ds of actionConfiguration.dataSources) {
+    /** Caveat: empty array in tags.in means "no document match" since no
+     * documents has any tags that is in the tags.in array (same for parents)*/
+    if (!config.DATASOURCE.filter.tags) {
+      config.DATASOURCE.filter.tags = {};
+    }
+    if (ds.filter.tags?.in) {
+      if (!config.DATASOURCE.filter.tags.in) {
+        config.DATASOURCE.filter.tags.in = [];
+      }
+      config.DATASOURCE.filter.tags.in.push(...ds.filter.tags.in);
+    }
+    if (ds.filter.tags?.not) {
+      if (!config.DATASOURCE.filter.tags.not) {
+        config.DATASOURCE.filter.tags.not = [];
+      }
+      config.DATASOURCE.filter.tags.not.push(...ds.filter.tags.not);
+    }
+    if (!config.DATASOURCE.filter.parents) {
+      config.DATASOURCE.filter.parents = {};
+    }
+    if (ds.filter.parents?.in) {
+      if (!config.DATASOURCE.filter.parents.in_map) {
+        config.DATASOURCE.filter.parents.in_map = {};
+      }
+      config.DATASOURCE.filter.parents.in_map[ds.dataSourceId] =
+        ds.filter.parents.in;
+    }
+    if (ds.filter.parents?.not) {
+      if (!config.DATASOURCE.filter.parents.not) {
+        config.DATASOURCE.filter.parents.not = [];
+      }
+      config.DATASOURCE.filter.parents.not.push(...ds.filter.parents.not);
+    }
+  }
+
+  // Handle timestamp filtering.
+  if (relativeTimeFrame) {
+    config.DATASOURCE.filter.timestamp = {
+      gt: timeFrameFromNow(relativeTimeFrame),
+    };
+  }
+
+  // Set top_k to 512 which is already a large number. We might want to bump to 1024.
+  config.DATASOURCE.top_k = 512;
```

Comment:
I'm missing something -- shouldn't this be a function of context size ?

Reply:
Oooh that's the total number of chunks we accept to process in the action actually ?

Reply:
Ok yes makes sense, we'll loop on batches of size N until we are done looking at all 512 chunks 👍

Reply:
Yes see design doc :+1: https://www.notion.so/dust-tt/Design-Doc-Process-Action-d0f9f14959b94bbfbe80f5deedf956ee?pvs=4